src/Model.php

```diff
@@ -1886,14 +1885,44 @@ public function action($mode, $args = [])
     /**
      * @param string $mode
      * @param array  $args
-     *
-     * @return Query
      */
-    public function toQuery($mode, $args = [])
+    public function toQuery($mode, $args = []): Persistence\AbstractQuery
```

Comment:
if this should be used a lot, we should redefined to `...$args` - or do we use/want to use assoc `$args` for something else than alias?

Reply:
My idea is to eventually directly use the query method on the model variable:
`$invoices->where('vendor_name', 'Test')->get();`

The model scope is automatically applied and the query methods are used to further adjust the query.

Reply:
wdym, your example does not use Query at all.

Reply:
Imagine below in Model with anotations to have hints in IDE. The methods we would like performing additional statements (like wrap `delete` in a transaction, hooks, etc) we can keep as method in Model.

```
    public function __call($method, $args)
    {
        return $this->toQuery($method, $args);
    }
```

Reply:
Above would not work because of dynamic methods so we can create a QueryTrait with the query methods

Reply:
I think we shifted a lot from original comment and you mix so many things at once
- (current/Model) toQuery methods are known before, there is not need to use dynamic methods
- annotations are used usually for extra info but still with impl. defined - the only exception are lazy properties, which can be not defined (only phpdoced) or unset at create time (but this is a little fragile as constructor can be skipped with refl)

Reply:
It is in the original PR description above but it is of course up for discussion.
And `Model::toQuery` will not be removed.

I suggest now I wrap up first stage here.
I will include all phpdocs and work on the comments on the code so far then we merge.
The `Model\QueryTrait` and `Model\QueryAggregatesTrait` will be easy to implement as either optional or standard functionality